### PR TITLE
tests can use the substitution option

### PR DESCRIPTION
**Pull request by dcrissman**
@bserdar would you please review and merge this PR? Thanks.

### CODE REVIEW

**Comment by dcrissman** [comment]
Seems to be failing because of ResultTranslatorToJsonTest#testTranslate_SimpleField_Date. I suspect this actually has to do with no longer standardizing to GMT. Needs more research, which I don't have time for right now.

**Comment by bserdar** [comment]
I see this:


testParse(com.redhat.lightblue.metadata.ldap.parser.LdapDataStoreParserTest)
 Time elapsed: 0.016 sec  <<< FAILURE!

org.junit.ComparisonFailure: expected:<[test]> but was:<[${ldap.database}]>

**Comment by dcrissman** [comment]
The error I got was local, I didn't even look at travis. I see that one, I will take a look.

**Comment by dcrissman** [comment]
Odd, I get a totally different error locally. I will file a bug with the details to be researched at another time. In the meantime, this now has a green light. Please review and merge.

File: lightblue-ldap-crud/src/test/java/com/redhat/lightblue/crud/ldap/translator/EntryTranslatorFromJsonTest.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ protected static Entry buildEntry(String fieldName, String metadataType, String
                 .replaceFirst("#fieldname", fieldName)
                 .replaceFirst("#value", crudValue);
 
-        EntityMetadata md = MetadataUtil.createEntityMetadata(LdapConstant.BACKEND, json(metadata), null, null);
+        EntityMetadata md = MetadataUtil.createEntityMetadata(LdapConstant.BACKEND, json(metadata, true), null, null);
```
Comment (bserdar) [inline lightblue-ldap-crud/src/test/java/com/redhat/lightblue/crud/ldap/translator/EntryTranslatorFromJsonTest.java]: This is likely to be 'false', unless you want property substitution in your  metadata.

File: lightblue-ldap-integration-test/src/test/java/com/redhat/lightblue/crud/ldap/ITCaseLdapCRUDController_InvalidMetadata_Test.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public void testMetadata_WithoutUniqueFieldDefined() throws Exception {
 
         //Remove the uid field from the fields definition
         String metadataWithoutUniqueField = AbstractJsonNodeTest.loadResource("./metadata/person-metadata.json").replaceFirst("\"uid\": \\{\"type\": \"string\"\\},", "");
-        JsonNode nodeWithoutUniqueField = json(metadataWithoutUniqueField);
+        JsonNode nodeWithoutUniqueField = json(metadataWithoutUniqueField, true);
```
Comment (bserdar) [inline lightblue-ldap-integration-test/src/test/java/com/redhat/lightblue/crud/ldap/ITCaseLdapCRUDController_InvalidMetadata_Test.java]: So is this.

Comment (dcrissman) [inline lightblue-ldap-integration-test/src/test/java/com/redhat/lightblue/crud/ldap/ITCaseLdapCRUDController_InvalidMetadata_Test.java]: Yes, I use substitution in the metadata. I use this for database and basedn values.

File: lightblue-ldap-crud/src/test/java/com/redhat/lightblue/crud/ldap/translator/EntryTranslatorFromJsonTest.java (hunk 1 above)
Comment (dcrissman) [inline lightblue-ldap-crud/src/test/java/com/redhat/lightblue/crud/ldap/translator/EntryTranslatorFromJsonTest.java]: The metadata here does not have substitution values in it. So I can revert this one.

**Pull request merged**